tools/hook-scripts/mailer/mailer.py

```diff
@@ -1442,32 +1443,48 @@ def repos_params(section_name, defaults):
         search_logmsg_re = re.compile(search_logmsg)
       else:
         search_logmsg_re = None
-
-      self._group_re.append((group,
-                             re.compile(for_paths),
-                             exclude_paths_re,
-                             params,
-                             search_logmsg_re))
+      suppress_if_match = getattr(sub, 'suppress_if_match', None)
+      if suppress_if_match == 'yes':
+        self._last_group_re.append((group,
+                                    re.compile(for_paths),
+                                    exclude_paths_re,
+                                    params,
+                                    search_logmsg_re,
+                                    True)) # Ignore if already matched
+      else:
+        self._group_re.append((group,
+                               re.compile(for_paths),
+                               exclude_paths_re,
+                               params,
+                               search_logmsg_re,
+                               False))
```

Comment:
Ugh, this is not nice. Remove the `if`.

```
self._last_group_re.append((group,
                            re.compile(for_paths),
                            exclude_paths_re,
                            params,
                            search_logmsg_re,
                            # Ignore if already matched
                            suppress_if_match == 'yes'))
```

Also, what guarantee is there that the true-value of `suppress_if_match` can only be `yes`? All I see is a new config option used in the code. Needs updated documentation in `mailer.conf.example`.

Reply:
Belay that suggested code change, I'm blind. The docco request stands, though.